cmd/thanos/main.go

```diff
@@ -212,6 +212,38 @@ func registerMetrics(mux *http.ServeMux, g prometheus.Gatherer) {
 	mux.Handle("/metrics", promhttp.HandlerFor(g, promhttp.HandlerOpts{}))
 }
 
+func registerHealthy(mux *http.ServeMux, logger log.Logger, component string) {
+	mux.HandleFunc("/-/healthy", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		if _, err := fmt.Fprintf(w, "Thanos %v is Healthy.\n", component); err != nil {
+			w.WriteHeader(http.StatusServiceUnavailable)
+			level.Error(logger).Log("msg", "Could not write health check response.")
+		}
+	})
+}
+
+func registerReady(mux *http.ServeMux, logger log.Logger, component string, readinessFunc func() (bool, error)) {
+	mux.HandleFunc("/-/ready", func(w http.ResponseWriter, r *http.Request) {
+		ready, err := readinessFunc()
+		if err != nil {
+			level.Error(logger).Log("msg", "Failed to get readiness status.")
+			ready = false
+		}
+
+		if ready {
+			w.WriteHeader(http.StatusOK)
+			if _, err := fmt.Fprintf(w, "Thanos %v is ready.\n", component); err != nil {
+				level.Error(logger).Log("msg", "Could not write readiness check response.")
+			}
+			return
+		}
+		w.WriteHeader(http.StatusServiceUnavailable)
+		if _, err := fmt.Fprintf(w, "Thanos %v is not ready.\n", component); err != nil {
+			level.Error(logger).Log("msg", "Could not write readiness check response.")
```

Comment:
Our convention is to have logs lowercase as well. Also worth to add `err` here for context. Same above.